pkg/pipeline/stream/source.go

```diff
@@ -175,6 +175,22 @@ func (n *SourceNode) registerMetricStatusHandler(msg *Message) {
 }
 
 func (n *SourceNode) Stop(ctx context.Context, reason error) error {
+	var err error
+	var stopExecuted bool
+	n.stopOnce.Do(func() {
+		// only execute stop once, more calls won't make a difference
+		err = n.stop(ctx, reason)
+		stopExecuted = true
+	})
+	if !stopExecuted {
+		n.logger.Warn(ctx).Msg("source connector stop already triggered, " +
+			"ignoring second stop request (if the pipeline is stuck, please " +
+			"report the issue to the Conduit team)")
+	}
```

Comment:
Maybe we return this as an error, so it's part of the API response?

Reply:
I was thinking about it when writing this. The drawback is that only the first request to stop the pipeline would succeed, all calls after that would fail unless the pipeline is started again. The UI would show this error, that didn't seem right 🤔
I agree that returning the error would give the caller the info what's happening, although then it would be nice if the UI caught the error and displayed it more as an info/warning rather than a full blown error.

Reply:
> although then it would be nice if the UI caught the error and displayed it more as an info/warning rather than a full blown error.

Agreed, it would be nice, and maybe not even too complex to be done in the front-end?

(But obviously, I don't consider this a blocker for the PR.)